### DISCUSSION TITLE
Migrating from custom_components install to HACS (or just upgrading)?

### DISCUSSION THREAD

**Discussion by owenfi** [PeteRager/lennoxs30#295] [General]
I have a custom_components install from a while ago. It's been working great, thank you!

I see that starting in October you are in the HACS default config. How do I port from my integration over to the HACS version?

If switching isn't recommended, how do I upgrade, just pull the latest release and drop it in? (while HA is shut down, I assume)

**Comment by PeteRager** [comment]
That's a great question.  I think it should just work.

HACS will want to deploy the files to the same custom_components directory and should just overwrite them.  An easy way to check is to look in the manifest.json file in custom_components/lennoxs30 as that contains the integration version which should be 2024.1.0 after HACs does the update.  If you do choose to go this route let me know if it works and I'll add a note to the docs.  

Alternatively, just grab the zip file from the latest release.  Unzip it and copy the contents of its custom_components/lennoxs30 directory into the HA custom_components directory (while HA is shutdown)

https://github.com/PeteRager/lennoxs30/releases/tag/2024.1.0

**Comment by owenfi** [comment]
Okay, I think good news here! (Something something...[procrastination always pays off now](https://despair.com/products/procrastination))

I had some unrelated issues around the time of the question and then got busy and didn't get around to it. I recently updated Home Assistant from 2023.9 -> 2024.2.2. Upon entering the HACS area I saw this:
<img width="616" alt="Screenshot 2024-02-18 at 10 45 31 PM" src="https://github.com/PeteRager/lennoxs30/assets/1295934/c83e1bec-94dc-483b-a2ca-ce513f1d77b3">

Similarly, if I went to HACS store > Integrations and searched Lennox I saw:
<img width="1098" alt="Screenshot 2024-02-18 at 10 46 58 PM" src="https://github.com/PeteRager/lennoxs30/assets/1295934/eaf961c2-7741-43af-9a16-c925ed544331">

Where the update button led to screenshot 1 as well.

I did the update, it took a few seconds and said it would apply after restart:
<img width="506" alt="Screenshot 2024-02-18 at 10 47 21 PM" src="https://github.com/PeteRager/lennoxs30/assets/1295934/42f99f72-0eb0-4359-a7e6-49e5cb910442">

I restarted Home Assistant. Spooked myself when I returned to HACS and didn't see it, but I guess that is just an "updates" section that it defaults to. The integration is of course still there and working fine! (and was upgraded as expected, confirmed in custom_components manifest.json)

Gotta say I love the new climate card as well. Subtle red glow when heating. Simple/fast +/- buttons for small changes. Mode control (might've been there before, but I might've missed it).

Thanks again, this integration brings me a ton of joy!
